scripts/scr_marine_struct/scr_marine_struct.gml

```diff
@@ -0,0 +1,386 @@
+			gear:{"armour":"Power Armour", "wep1":"Power Sword", "wep2":"Storm Shield"}
+	},
+	"ork_sniper":{
+			title : "Ork Sniper",
+			strength:5,
+			constitution:9,
+			dexterity:8,
+			intelligence:2,
+			wisdom:3,
+			charisma :4,
+			religion : "gorkamorka",
+			piety : 20,
+			technology :2,
+			luck :4,
+			skills: {},	
+			gear:{"armour":"Power Armour", "wep1":"Power Sword", "wep2":"Storm Shield"}
+	},
+	"flash_git":{
+			title : "Flash Git",
+			strength:8,
+			constitution:13,
+			dexterity:6,
+			intelligence:2,
+			wisdom:3,
+			charisma :4,
+			religion : "gorkamorka",
+			piety : 20,
+			technology :2,
+			luck :4,
+			skills: {},	
+			gear:{"armour":"Power Armour", "wep1":"Power Sword", "wep2":"Storm Shield"}
+	}
+}
+function TTRPG_stats(faction, comp, mar, class = "marine") constructor{
+	
+	// used both to load unit data from save and to add preset base_stats
+	static load_json_data = function(data){
+		 var names = variable_struct_get_names(data);
+		 for (var i = 0; i < array_length(names); i++) {
+            variable_struct_set(self, names[i], variable_struct_get(data, names[i]))
+        }
+	}	
+	traits = [];			//marine trait list
+	company = comp;			//marine company
+	marine_number = mar;	//marine number in company
+	allegiance =faction;	//faction alligience defaults to the chapter
+	gear = {"armour":0, "wep1":0, "wep2":0, "mobi":0}; //item used to re-equip marine
+	
+	//adds a trait to a marines trait list
+	static add_trait = function(trait){
+			var selec_trait = global.trait_list[$ trait];
+			var edits = variable_struct_get_names(selec_trait);
+			
+			//loop over stats and add stats where needed
+			var stats = ["constitution", "strength", "luck", "dexterity", "wisdom", "piety", "charisma", "technology"]
+			for (var stat_iter =0; stat_iter <array_length(stats);stat_iter++;){
+				if (array_contains(edits ,stats[stat_iter])){
+					 variable_struct_set(self,stats[stat_iter],  (variable_struct_get(self, stats[stat_iter])+  variable_struct_get(selec_trait, stats[stat_iter])));
+				}
+			}
+			//max_health = 100 * (1+((constitution - 20)*0.05));
+			array_push(traits, trait);
+	}
+	
+	//takes dict and plumbs dict values into unit struct
+	if (array_contains(variable_struct_get_names(global.base_stats), class)){
+		load_json_data(global.base_stats[$ class]);
+	}
+	
+
+	switch class{
+		case "marine":				//basic marine class //adds specific mechanics not releveant to most units
+			if (faction ="chapter"){
+				allegiance = global.chapter_name;
+			}
+			if (instance_exists(obj_controller)){
+				role_history = [[obj_ini.role[company,marine_number], obj_controller.turn]]; //marines_promotion and demotion history
+				marine_ascension = obj_controller.turn; // on what day did turn did this marine begin to exist
+			} else {
+				role_history = [[obj_ini.role[company,marine_number], "pre_game"]];
+				marine_ascension = "pre_game"; // on what day did turn did this marine begin to exist
+
+			}
+			if (irandom(100)>99){
```

Comment:
`irandom(100)` will return values from 0 to 100 _inclusive_, so this actually has a 99/101 chance of triggering, instead of 99/100

Reply:
altered to correct